test/e2e/scrapeconfig_test.go

```diff
@@ -2119,3 +2122,148 @@ var LightSailSDTestCases = []scrapeCRDTestCase{
 		expectedError: true,
 	},
 }
+
+var OVHCloudSDTestCases = []scrapeCRDTestCase{
+	{
+		name: "Valid OVHCloudSDConfig",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			OVHCloudSDConfigs: []monitoringv1alpha1.OVHCloudSDConfig{
+				{
+					ApplicationKey:    "valid-app-key",
+					ApplicationSecret: v1.SecretKeySelector{Key: "valid-secret-key"},
+					ConsumerKey:       v1.SecretKeySelector{Key: "valid-consumer-key"},
+					Service:           "VPS",
```

Comment:
(nit) instead of using "VPS", try using the constants defined [here](https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go#L1065-L1072).
```suggestion
					Service:           monitoringv1alpha1.VPS,
```

Reply:
We could improve the naming of the constants too. Maybe `OVHServiceVPS` and `OVHServiceDedicatedServer`.